Add release notes for 0.215

nezihyigitbasi opened the pull request:
Per our offline discussion with @zaynoon, here is a follow up PR for https://github.com/prestodb/presto/pull/12076 with comments addressed.

Fixes https://github.com/prestodb/presto/issues/11961

rongrong commented on presto-docs/src/main/sphinx/release/release-0.215.rst:
```diff
@@ -0,0 +1,57 @@
+---------------
+
+* Fix reporting of the processed input data for source stages in ``EXPLAIN ANALYZE``.
+* Fix bug where calculation of estimated block size can be quadratic. This fixes CPU waste for queries
```
`This fixes CPU waste for a small number of queries`

electrum commented on presto-docs/src/main/sphinx/release/release-0.215.rst:
```diff
@@ -0,0 +1,57 @@
+* Add :func:`beta_cdf` and :func:`inverse_beta_cdf` functions.
+* Split the reporting of raw input data and processed input data for source operators.
+* Presto no longer reports raw input data for ``Values``, ``Local Exchange``, and ``Local Merge Sort`` operators.
+* Simplify ``EXPLAIN (TYPE IO)`` output when there are too many discrete components. This avoids large output
```
This item is good but the line is too long.

electrum commented on presto-docs/src/main/sphinx/release/release-0.215.rst:
```diff
@@ -0,0 +1,57 @@
+* Add input validation checks to :func:`ST_LineString` to conform with the specification.
+* Add :func:`beta_cdf` and :func:`inverse_beta_cdf` functions.
+* Split the reporting of raw input data and processed input data for source operators.
+* Presto no longer reports raw input data for ``Values``, ``Local Exchange``, and ``Local Merge Sort`` operators.
```
"Presto no longer reports" is rather vague. Please clarify what this is referring to.

nezihyigitbasi replied:
/cc @sopel39

nezihyigitbasi replied:
Looking at the relevant commits I rephrased this as:

```
* Presto no longer collects and reports the raw input data statistics for the ``Values``,
  ``Local Exchange``, and ``Local Merge Sort`` operators.
```

sopel39 replied:
lgtm

electrum commented on presto-docs/src/main/sphinx/release/release-0.215.rst:
```diff
@@ -0,0 +1,57 @@
+* Fix reporting of the processed input data for source stages in ``EXPLAIN ANALYZE``.
+* Fix bug where calculation of estimated block size can be quadratic. This fixes CPU waste for queries
+* Fix regression in 0.214 that could cause queries to produce incorrect results due to a bug in lazy hash table build for map types.
+* Improve spatial join performance by accelerating build-side geometries.
```
Let's move all of the geospatial changes to a "Geospatial Changes" section as we have done in the past.

electrum commented on presto-docs/src/main/sphinx/release/release-0.215.rst:
```diff
@@ -0,0 +1,57 @@
+Web UI Changes
+--------------
+
+* Fix ``Capture Snapshot`` button for showing current thread stacks.
```
Let's use italics for this since it's not code:
```
* Fix *Capture Snapshot* button for showing current thread stacks.

electrum commented on presto-docs/src/main/sphinx/release/release-0.215.rst:
```diff
@@ -0,0 +1,57 @@
+--------------
+
+* Fix ``Capture Snapshot`` button for showing current thread stacks.
+* Fix drop-down to show stage skew bar chart.
```
Which drop-down? Which page?

nezihyigitbasi replied:
/cc @raghavsethi

electrum commented on presto-docs/src/main/sphinx/release/release-0.215.rst:
```diff
@@ -0,0 +1,57 @@
+
+* Fix ``Capture Snapshot`` button for showing current thread stacks.
+* Fix drop-down to show stage skew bar chart.
+* Fix expand dropdown for stage skew component.
```
Use consistent spelling across items: either drop-down or dropdown

Which drop-down? Is this a duplicate of the previous item?

nezihyigitbasi replied:
I think this is the same as "Fix expand dropdown for stage skew component." I agree we need info about which page, etc. this is affecting. /cc @raghavsethi

electrum commented on presto-docs/src/main/sphinx/release/release-0.215.rst:
```diff
@@ -0,0 +1,57 @@
+* Fix ``Capture Snapshot`` button for showing current thread stacks.
+* Fix drop-down to show stage skew bar chart.
+* Fix expand dropdown for stage skew component.
+* Improve the performance of the thread snapshot component in the worker status page.
```
in the -> on the

electrum commented on presto-docs/src/main/sphinx/release/release-0.215.rst:
```diff
@@ -0,0 +1,57 @@
+* Fix drop-down to show stage skew bar chart.
+* Fix expand dropdown for stage skew component.
+* Improve the performance of the thread snapshot component in the worker status page.
+* Make the reporting of "Cumulative Memory" usage consistent on query list and query details pages.
```
Use italics
```
* Make the reporting of *Cumulative Memory* usage consistent on query list and query details pages.

electrum commented on presto-docs/src/main/sphinx/release/release-0.215.rst:
```diff
@@ -0,0 +1,57 @@
+PostgreSQL Changes
+------------------
+
+* Upgrade PostgreSQL JDBC driver to 42.2.5 (from 42.1.4).
```
This is an implementation detail. If there are user visible changes, explain in those terms, otherwise remove this.

nezihyigitbasi replied:
/cc @findepi

findepi replied:
i'm ok removing. They claim some security changes, but I wouldn't be able to name them.

nezihyigitbasi replied:
Removed.

nezihyigitbasi commented:
Per our offline discussion with @rongrong, removed the item:

```
* Fix bug where calculation of estimated block size can be quadratic. This fixes CPU waste for queries
```

Also, I guess the change about the skew component in the web ui changes is the following commit:
https://github.com/prestodb/presto/commit/c5451f115a8fd06eebc2a0269d61c9bc18d4fd9c. So, I just updated that item to say that the fix affects the query details page.
